src/Features/CSharp/Portable/ConvertTupleToStruct/CSharpConvertTupleToStructCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.ConvertTupleToStruct;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertTupleToStruct
+{
+    [ExtensionOrder(Before = PredefinedCodeRefactoringProviderNames.IntroduceVariable)]
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(PredefinedCodeRefactoringProviderNames.ConvertTupleToStruct)), Shared]
+    internal class CSharpConvertTupleToStructCodeRefactoringProvider :
+        AbstractConvertTupleToStructCodeRefactoringProvider<
+            ExpressionSyntax,
+            NameSyntax,
+            IdentifierNameSyntax,
+            LiteralExpressionSyntax,
+            ObjectCreationExpressionSyntax,
+            TupleExpressionSyntax,
+            ArgumentSyntax,
+            TupleTypeSyntax,
+            TypeDeclarationSyntax,
+            NamespaceDeclarationSyntax>
+    {
+        protected override ObjectCreationExpressionSyntax CreateObjectCreationExpression(
+            NameSyntax nameNode, SyntaxToken openParen, SeparatedSyntaxList<ArgumentSyntax> arguments, SyntaxToken closeParen)
+        {
```

Comment:
> { [](start = 7, length = 2)

nit: you may want to convert this to expression body

Reply:
my informal rule is that i generally don't like that if hte header of the method is more than one line.  it looks a little too cluttered to me...